packages/now-node/src/index.ts

```diff
@@ -474,27 +478,49 @@ async function doTypeCheck({
   // In order to type-check a single file, a temporary tsconfig
   // file needs to be created that inherits from the base one :(
   // See: https://stackoverflow.com/a/44748041/376773
-  const id = Math.random()
-    .toString(32)
-    .substring(2);
-  const tempConfigName = `.tsconfig-${id}.json`;
+  const absoluteEntrypoint = join(workPath, entrypoint);
   const projectTsConfig = await walkParentDirs({
     base: workPath,
     start: join(workPath, dirname(entrypoint)),
     filename: 'tsconfig.json',
   });
+  const sha = createHash('sha256')
+    .update(absoluteEntrypoint)
+    .update(projectTsConfig || '')
+    .digest('hex');
+  const tempConfigName = `vercel-node-${sha}.json`;
+  const tempConfigPath = join(TMP, tempConfigName);
   const tsconfig = {
     extends: projectTsConfig || undefined,
-    include: [entrypoint],
+    include: [absoluteEntrypoint],
   };
-  await fsp.writeFile(tempConfigName, JSON.stringify(tsconfig));
+
+  console.error({
+    absoluteEntrypoint,
+    sha,
+    tempConfigName,
+    tempConfigPath,
+    tsconfig,
+  });
+
+  try {
+    await fsp.writeFile(tempConfigName, JSON.stringify(tsconfig), {
+      flag: 'wx',
+    });
+  } catch (err) {
+    console.error(err);
+    // Don't throw if the file already exists
+    if (err.code !== 'EEXIST') {
+      throw err;
+    }
+  }
 
   const child = spawn(
     process.execPath,
     [
       tscPath,
       '--project',
-      tempConfigName,
+      tempConfigPath,
```

Comment:
I wonder if we could reuse the path instead of creating a new one each time? We have the `meta.devCachDir` which is already utilized by Go and Python. Maybe we can use it here?

Reply:
The PR has been updated to store them in the `devCacheDir`. For example: `.now/cache/tsconfig-b45d22214f35c8ac6110f2b4045145343afa1db3bed866d4643614395c496b2d.json`.